### PR TITLE
Make datapath configurable through application.conf

### PR DESCRIPTION
**Pull request by mlachkar**
- make prostgresql write locally in .postgresql directory:
it will allow not to insert data everytime locally

### CODE REVIEW

File: project/Deployment.scala [hunk 1]
```diff
@@ -128,10 +128,11 @@ class Deployment(
...
           |  -Dsentry.release=$version \\
-          |  -Dapi.port=$port \\
-          |  /home/$userName/scaladex-contrib \\
-          |  /home/$userName/scaladex-index \\
-          |  /home/$userName/scaladex-credentials \\
+          |  -Dapp.port=$port \\
+          |  -Dapp.env=prod \\
+          |  -Ddata-paths.contrib=/home/$userName/scaladex-contrib \\
+          |  -Ddata-paths.index=/home/$userName/scaladex-index \\
+          |  -Ddata-paths.credentials=/home/$userName/scaladex-credentials \\
```
Comment (adpi2) [inline project/Deployment.scala:135]: Can we remove that and put them in `~/scaladex-credentials/app.conf` ?

File: data/src/main/scala/ch.epfl.scala.index.data/DataPaths.scala [hunk 2]
```diff
@@ -70,48 +71,42 @@ object LocalPomRepository {
 
 object DataPaths {
 
-  private val log = LoggerFactory.getLogger(getClass)
-
   private val base = build.info.BuildInfo.baseDirectory.toPath.getParent
```
Comment (adpi2) [inline data/src/main/scala/ch.epfl.scala.index.data/DataPaths.scala:74]: Do we still need that?

File: data/src/main/scala/ch.epfl.scala.index.data/DataPaths.scala [hunk 3]
```diff
@@ -70,48 +71,42 @@ object LocalPomRepository {
...
-          Paths.get(index),
-          Paths.get(credentials)
-        )
-      case _ => {
-        (
-          defaultContrib,
-          defaultIndex,
-          defaultCredentials
-        )
+  base.resolve(Paths.get("scaladex-credentials"))
```
Comment (adpi2) [inline data/src/main/scala/ch.epfl.scala.index.data/DataPaths.scala:75]: and that?

File: server/src/main/resources/reference.conf [hunk 4]
```diff
@@ -10,11 +10,18 @@ org.scala_lang.index.server {
...
+app {
+   env = "local"
    endpoint = "localhost"
    port = 8082
 }
 
+data-paths {
+  contrib = "scaladex-contrib"
+  index = "scaladex-small-index"
+  credentials = "scaladex-credentials"
```
Comment (adpi2) [inline server/src/main/resources/reference.conf:22]: locally we were using `../scaladex-contrib`?

File: project/Deployment.scala (hunk 1 above)
Comment (mlachkar) [reply project/Deployment.scala:135]: yes! next step . just to make this release not dependent on anything

**Pull request merged**